src/sql/sql-query/sql-query.spec.ts

```diff
@@ -3022,3 +3022,162 @@ describe('expressions with union clause', () => {
+        "innerSpacing": Object {
+          "postExplain": "",
+          "postFrom": " ",
+          "postLimitKeyword": "",
+          "postQuery": "",
+          "postSelect": " -- some comment 
+        ",
+          "postSelectDecorator": "",
+          "postSelectValues": " ",
+          "postUnionKeyword": "",
+          "postWith": "",
+          "postWithQuery": "",
+          "preGroupByKeyword": "",
+          "preHavingKeyword": "",
+          "preLimitKeyword": "",
+          "preQuery": "",
+          "preUnionKeyword": "",
+          "preWhereKeyword": "",
+        },
+        "limitKeyword": undefined,
+        "limitValue": undefined,
+        "orderByKeyword": undefined,
+        "orderBySeparators": undefined,
+        "orderByUnits": undefined,
+        "selectDecorator": "",
+        "selectKeyword": "Select",
+        "selectSeparators": Array [],
+        "selectValues": Array [
+          SqlRef {
+            "innerSpacing": Object {},
+            "name": "column",
+            "namespace": undefined,
+            "namespaceQuotes": undefined,
+            "quotes": "",
+            "type": "ref",
+          },
+        ],
+        "tableSeparators": Array [],
+        "tables": Array [
+          SqlRef {
+            "innerSpacing": Object {},
+            "name": "table",
+            "namespace": undefined,
+            "namespaceQuotes": undefined,
+            "quotes": "",
+            "type": "ref",
+          },
+        ],
+        "type": "query",
+        "unionKeyword": undefined,
+        "unionQuery": undefined,
+        "whereExpression": undefined,
+        "whereKeyword": undefined,
+        "withKeyword": undefined,
+        "withSeparators": undefined,
+        "withUnits": undefined,
+      }
+    `);
+  });
+
+  it('two comments', () => {
+    const sql = `Select --some comment
+    --some comment
+  column from table`;
+
+    expect(parser(sql).toString()).toMatchInlineSnapshot(`
+      "Select --some comment
+          --some comment
+        column from table"
+    `);
+  });
+
+  it('comment on new line', () => {
+    const sql = `Select
+    -- some comment
+  column from table`;
+
+    expect(parser(sql).toString()).toMatchInlineSnapshot(`
+      "Select
+          -- some comment
+        column from table"
+    `);
+  });
+
+  it('comment containing hyphens', () => {
+    const sql = `Select
+    -- some--comment
+  column from table`;
+
+    expect(parser(sql).toString()).toMatchInlineSnapshot(`
+      "Select
+          -- some--comment
+        column from table"
+    `);
+  });
+
+  it('comment with no space', () => {
+    const sql = `Select --some comment
+  column from table`;
+
+    expect(parser(sql).toString()).toMatchInlineSnapshot(`
+      "Select --some comment
+        column from table"
+    `);
+  });
+  it('comment with non english', () => {
+    const sql = `Select --Здравствуйте
+  column from table`;
+
+    expect(parser(sql).toString()).toMatchInlineSnapshot(`
```

Comment:
all of these are expected to give you the input back right? Could it just be `expect(parser(sql).toString()).toEqual(sql)` ?

Reply:
Yeah it could be, I can change that